### PR TITLE
App Submission : Homebox

### PR DESCRIPTION
**Pull request by Xosten**
# App Submission

### 256x256 SVG icon

https://hay-kot.github.io/homebox/assets/img/lilbox.svg

### Gallery images

Homebox1
Homebox2
Homebox3
Homebox4
Homebox5

### I have tested my app on:
- [ ] [Umbrel dev environment](https://github.com/getumbrel/umbrel-dev)
- [x] [Umbrel OS on a Raspberry Pi 4](https://github.com/getumbrel/umbrel-os)
- [ ] [Custom Umbrel install on Linux](https://github.com/getumbrel/umbrel#-installation)

### CODE REVIEW

File: homebox/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+version: "3.7"
+
+services:
+
+  app_proxy:
+    environment:
+      APP_HOST: homebox_server_1
```
Comment (nmfretz) [inline homebox/docker-compose.yml:7]: For single container apps we have been naming the container `web` to standardize things.

```suggestion
      APP_HOST: homebox_web_1
```

File: homebox/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      APP_PORT: 7745
+    
+  server:
```
Comment (nmfretz) [inline homebox/docker-compose.yml:10]:
```suggestion
  web:
```

File: homebox/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    image: ghcr.io/hay-kot/homebox:v0.9.1@sha256:93eb59c89dc59cf2a6bd3943828ecc30f5a8c8f34fd844c79f54801e914ea64e
+    restart: on-failure
+    environment:
+      - PUID=1000
+      - PGID=1000
```
Comment (nmfretz) [inline homebox/docker-compose.yml:15]: It doesn't look like the homebox image supports these environment variables. To make sure that the data directory gets created with the correct non-root user permissions you can add `user: 1000:1000` above or below the `environment` section. See this example from the Lightning  Node app:

https://github.com/getumbrel/umbrel-apps/blob/6841e3b74b7cf328ce4e8e65be7f5b140533440f/lightning/docker-compose.yml#L43-L48

**Review by nmfretz — COMMENTED** [review]
Thanks for this submission @Xosten! This is almost ready to go. See my review for minor changes.

I will get an app icon and gallery images prepared.

File: homebox/docker-compose.yml (hunk 1 above)
Comment (Xosten) [reply homebox/docker-compose.yml:7]: Thanks for bringing this to my attention, been longtime since I made an app suggestion ^^

**Comment by Xosten** [comment]
> Thanks for this submission @Xosten! This is almost ready to go. See my review for minor changes.
> 
> I will get an app icon and gallery images prepared.

Thanks Nathan for your feedback, it has been a while since I made an app submission, sorry for the small mistakes 😉.

One thing I'm doubting is "Productivity" as a category... I would maybe see "Home" or something related as a category and maybe add at the same time "Homeassistant" to it. What do you think?

**Comment by nmfretz** [comment]
Whoops, just tested and I think I advised you incorrectly re the env vars. I think those may be required for SQLite. Are you okay if I push some changes to your branch so I can troubleshoot a bit?

> One thing I'm doubting is "Productivity" as a category... I would maybe see "Home" or something related as a category and maybe add at the same time "Homeassistant" to it. What do you think?

oh, ya good suggestion. Thanks for that. Let's leave as Productivity for now and we will look into revamping our categories as more apps are submitted.

**Comment by Xosten** [comment]
> Whoops, just tested and I think I advised you incorrectly re the env vars. I think those may be required for SQLite. Are you okay if I push some changes to your branch so I can troubleshoot a bit?

Sure go ahead 😉, I'm currently not available to test it on my side 😕

> oh, ya good suggestion. Thanks for that. Let's leave as Productivity for now and we will look into revamping our categories as more apps are submitted.

Sounds like a plan!

**Comment by nmfretz** [comment]
Ah, I see what's happening now. We do indeed want to get rid of the env vars `PUID=1000` and `PGID=1000`. 

The remaining issue is that the current [Dockerfile](https://github.com/hay-kot/homebox/blob/main/Dockerfile) used by Homebox results in the container being run as root, which we really want to avoid for Umbrel apps for security reasons. I see there is an open PR https://github.com/hay-kot/homebox/pull/372 for publishing a non-root image, which we should get soon. Once this is done, this should be good to go.

**Comment by nmfretz** [comment]
I commented out `user: 1000:1000` for now, until a new non-root image is pushed.

I also added the `data` directory so that this will exist and be owned by user 1000, instead of being created by docker with root permissions.

As-is, the app works well, we just need to solve the root permissions issue.

**Comment by Xosten** [comment]
Hey Nathan! Thanks for your investigation! 

I have subscribed to the open PR you mentioned, hopefully they add it soon 😄 !  In the meantime, it allows your team to create the App Store assets without pressure 😊. 

Cheers!

**Comment by nmfretz** [comment]
Thanks Xosten!

**Comment by nmfretz** [comment]
Any news on this @Xosten? Looks like maybe the -rootless tag never got pushed

**Comment by nmfretz** [comment]
@Xosten FYI I just pinged the PR again: https://github.com/hay-kot/homebox/pull/372#issuecomment-1661073037

**Comment by Xosten** [comment]
> @Xosten FYI I just pinged the PR again: [hay-kot/homebox#372 (comment)](https://github.com/hay-kot/homebox/pull/372#issuecomment-1661073037)

Thanks @nmfretz, I totally forgot to follow up with the PR, sorry about that

**Comment by nmfretz** [comment]
oh no worries at all! I was just letting you know and posting so we can keep track of things.

**Comment by nmfretz** [comment]
@Xosten I just tested out the rootless test-image. The app is working correctly, and `user: 1000:1000` results in correct permissions.

While we await kay-hot publishing a versioned image I will get the @surajmahraj to help us create an icon and gallery images.

**Comment by nmfretz** [comment]
@Xosten this is going live to the app store! Thank you for all your work on this. I have locked the image to this specific digest for now `sha256:1ec765caa21ad37a1fe64eaed2b10b1652169389a275c914eab71e6fd10232b9` (nightly rootless image). When kay-hot pushes a versioned tag we can update.

Here are the gallery assets ⬇

gallery images:





icon:
icon

**Comment by Xosten** [comment]
@nmfretz Thank you for your help too man! Those images are sick! Really happy to see this going to the Appstore :D

**Pull request merged**